securedrop/journalist_app/__init__.py

```diff
@@ -119,39 +118,13 @@ def _handle_http_exception(
         template_filters.html_datetime_format
     app.jinja_env.add_extension('jinja2.ext.do')
 
-    @app.before_first_request
-    def expire_blacklisted_tokens() -> None:
-        cleanup_expired_revoked_tokens()
-
     @app.before_request
     def update_instance_config() -> None:
         InstanceConfig.get_default(refresh=True)
 
     @app.before_request
     def setup_g() -> 'Optional[Response]':
         """Store commonly used values in Flask's special g object"""
-        if 'expires' in session and datetime.now(timezone.utc) >= session['expires']:
-            session.clear()
-            flash(gettext('You have been logged out due to inactivity.'),
```

Comment:
As briefly discussed with @lsd-cat today: As I understand, we're no longer able to show specific error message because we no longer have information about the session after it expires. So the current behavior of the PR is to not show any error at all. Is that correct? If so, I would suggest that we may still want to show an error message if we can't validate a session present in the cookie, just a more generic one, to give a hint to the user that the behavior is expected (and to aid with troubleshooting).

Reply:
Right. And we have no knowledge whether the session was legit and expired, or just a bogus session.

We could do something like if there's no valid session, seeing if they tried to pass a session cookie in the request, and if so, flashing a generic error, "Your session has expired or is invalid. Please log in again."

Reply:
We have indeed a condition in which we could assume that the session was valid, but got destroyed server-side either due to timeout, user/admin password change or account deletion. The case is when the session cookie has a valid signature for the session_id, but then the session is not in Redis. Not sure if it is better to use this condition, or the more generic has a session cookie to display the message.

Reply:
Implemented that case here https://github.com/freedomofpress/securedrop/pull/6403/commits/22b409217d1cc3b298cba2a0dd7d4940601939cf
The onyl thing is that at that point in time we do not have knowledge of the locale anymore. So either we use always english, or we try to get the request accept-language and decide based only on that.